modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -437,4 +439,33 @@ public void onShutdown() {
     public MvPartitionStorage getMvStorage() {
         return storage.getStorage();
     }
+
+    /**
+     * Handler for the {@link BuildIndexCommand}.
+     *
+     * @param cmd Command.
+     * @param commandIndex RAFT index of the command.
+     * @param commandTerm RAFT term of the command.
+     */
+    void handleBuildIndexCommand(BuildIndexCommand cmd, long commandIndex, long commandTerm) {
+        // Skips the write command because the storage has already executed it.
```

Comment:
I wonder, why do we have this check in every single method, instead of making it once in "onWrite". Any ideas?

Reply:
No ideas.

Reply:
Maybe you can move it outside :wink:

Reply:
I'm a little worried that I might break something, perhaps I'll fix it in a separate ticket along with other refactorings.
https://issues.apache.org/jira/browse/IGNITE-19113